libr/anal/p/anal_x86_cs.c

```diff
@@ -461,7 +451,82 @@ Sets the byte in the operand to 1 if the Sign Flag is not equal
 	case X86_INS_CMOVNS:
 	case X86_INS_CMOVO:
 	case X86_INS_CMOVP:
-	case X86_INS_CMOVS:
+	case X86_INS_CMOVS: {
+		const char *conditional;
+		char *src = getarg (&gop, 1, 0, NULL);
+		char *dst = getarg (&gop, 0, 1, NULL);
+		switch (insn->id) {
+		case X86_INS_CMOVA:
+			// mov if CF = 0 *AND* ZF = 0
+			conditional = "cf,zf,|,!";
+			break;
+		case X86_INS_CMOVAE:
+			// mov if CF = 0
+			conditional = "cf,!";
+			break;
+		case X86_INS_CMOVB:
+			// mov if CF = 1
+			conditional = "cf";
+			break;
+		case X86_INS_CMOVBE:
+			// mov if CF = 1 *OR* ZF = 1
+			conditional = "cf,zf,|";
+			break;
+		case X86_INS_CMOVE:
+			// mov if ZF = 1
+			conditional = "zf";
+			break;
+		case X86_INS_CMOVG:
+			// mov if ZF = 0 *AND* SF = OF
+			conditional = "zf,!,sf,of,==,&";
+			break;
+		case X86_INS_CMOVGE:
+			// mov if SF = OF
+			conditional = "sf,of,==";
+			break;
+		case X86_INS_CMOVL:
+			// mov if SF != OF
+			conditional = "sf,of,!=";
+			break;
+		case X86_INS_CMOVLE:
+			// mov if ZF = 1 *OR* SF != OF
+			conditional = "zf,sf,of,!=,|";
+			break;
+		case X86_INS_CMOVNE:
+			// mov if ZF = 0
+			conditional = "zf,!";
+			break;
+		case X86_INS_CMOVNO:
+			// mov if OF = 0
+			conditional = "of,!";
+			break;
+		case X86_INS_CMOVNP:
+			// mov if PF = 0
+			conditional = "pf,!";
+			break;
+		case X86_INS_CMOVNS:
+			// mov if SF = 0
+			conditional = "sf,!";
+			break;
+		case X86_INS_CMOVO:
+			// mov if OF = 1
+			conditional = "of";
+			break;
+		case X86_INS_CMOVP:
+			// mov if PF = 1
+			conditional = "pf";
+			break;
+		case X86_INS_CMOVS:
+			// mov if SF = 1
+			conditional = "sf";
+			break;
+		}
+		if (src && dst) {
+			esilprintf (op, "%s,?{,%s,%s,}", conditional, src, dst);
```

Comment:
what if someone changes one switch and not the other? maintainability and being non error-prone is important

> On 16 May 2016, at 00:11, Sushant Dinesh <notifications@github.com> wrote:
> 
> In libr/anal/p/anal_x86_cs.c <https://github.com/radare/radare2/pull/4884#discussion_r63298824>:
> 
> > +			break;
> > +		case X86_INS_CMOVO:
> > +			// mov if OF = 1
> > +			conditional = "of";
> > +			break;
> > +		case X86_INS_CMOVP:
> > +			// mov if PF = 1
> > +			conditional = "pf";
> > +			break;
> > +		case X86_INS_CMOVS:
> > +			// mov if SF = 1
> > +			conditional = "sf";
> > +			break;
> > +		}
> > +		if (src && dst) {
> > +			esilprintf (op, "%s,?{,%s,%s,}", conditional, src, dst);
> @radare <https://github.com/radare> It cannot. It has to belong to one of those cases. See the bigger switch case matched above.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub <https://github.com/radare/radare2/pull/4884/files/0736f133a852587249713fc79539d5d23f897f96#r63298824>